src/main/native/com/code_intelligence/jazzer/driver/fuzzed_data_provider.cpp

```diff
@@ -96,9 +98,10 @@ ConsumeIntegralArray(JNIEnv &env, jobject self, jint max_length) {
   const auto *dataPtr =
       reinterpret_cast<const uint8_t *>(env.GetLongField(self, gDataPtrField));
   jint remainingBytes = env.GetIntField(self, gRemainingBytesField);
+  jint requested_bytes = static_cast<jint>(sizeof(T)) * max_length;
 
   jint max_num_bytes =
-      std::min(static_cast<jint>(sizeof(T)) * max_length, remainingBytes);
+      std::max(std::min(requested_bytes, remainingBytes), static_cast<jint>(0));
```

Comment:
While this fixes a crash, it can result in somewhat weird behavior: If `max_length` grows, `max_num_bytes` may shrink.

Instead, we could perform the multiplication above on `uint64_t` to prevent an overflow and then truncate to `jint` here. This would result in saturation at `remainingBytes` rather than fluctuation. What do you think?

Reply:
I'm not sure that using a `uint64_t` here would help much but I may not know enough about C++ conversion options. I tried `static_cast<int64_t>(<a uint64_t bigger than int64_t::max>)` and it resulted in a negative. Is there a saturating conversion function available?
Like
```c++
    uint64_t bytes = static_cast<uint64_t>(std::numeric_limits<int32_t>::max()) + 12340;
    int32_t bytes_int = static_cast<int32_t>(bytes);
    std::cout << "bytes: " << bytes << std::endl;
    std::cout << "int  : " << bytes_int << std::endl;
```

Otherwise, I would probably keep this `jint` territory and do a check if it's negative.

Reply:
Sorry, my comment wasn't fully accurate: You have to perform the multiplication *and* `std::min` on `uint64_t`, and only then truncate to `jint`.

Reply:
Pushed a fixed version. If you think that's good, I'll fix up the commits

Reply:
I left one comment, but if I missed something about that, this version is fine too.